include/deal.II/base/discrete_time.h

```diff
@@ -71,19 +83,43 @@ class DiscreteTime
   /**
    * Return the end of the time interval.
    * The final time step ends exactly at this point. This exact floating-point
-   * equality is very important because it allows us to use the expression
-   * <code>time.get_current_time() != time.get_end_time()</code> as the
-   * conditional statement in a for loop to check if the end time is reached.
+   * equality is very important because it allows us to equality-compare
+   * current time with end time and decide whether we have reached the end of
+   * the simulation.
    */
   double
   get_end_time() const;
 
+  /**
+   * Return whether no step has taken place yet.
+   */
+  bool
+  is_at_start() const;
+
+  /**
+   * Return whether time has reached the end time.
+   */
+  bool
+  is_at_end() const;
+
   /**
    * Return the size of the step from current time step to the next.
    */
   double
   get_next_step_size() const;
 
+  /**
+   * Return the step size of the previous step.
+   */
+  double
+  get_previous_step_size() const;
+
+  /**
+   * Return the number of times the simuation time is incremented.
+   */
+  unsigned int
+  get_step_number() const;
```

Comment:
Couldn't this just get called the `iteration`?

Reply:
I know that sometimes *step number* and *iteration* are used interchangeably, but I believe they are not the same concept. *Iteration* implies that we are starting with an approximate solution and iterating multiple times until we *converge* to the true solution. That is why we use iteration to describe a Newton-Raphson algorithm. However, time stepping in a time-dependent problem does not mean that we are iterating to converge to a perfect solution. The solution that we obtain for each step is accurate for that specific time. We are just incrementing the time and recalculating the correct solution for the new time.

There is one area where *iteration* and *step number* refer to the same thing. When we are solving a nonlinear PDE with boundary conditions that are large (in terms of their norm). We are interested to compute the solution in one step but we can't do it because of non-linearity. We introduce the concept of pseudo time and increase the norm of the BC in small increments and solve the system until we reach the final large BC. In this case, we are *iterating* by *stepping* forward in pseudo time.

Reply:
It's an interesting distinction -- quite useful to verbalize this and settle on a single term!

Reply:
Indeed, thank you for the enlightening response. I'm perfectly OK with leaving this as it is, and it certainly helps keep distinct what's happening per timestep and what's happening at the (non)linear solver level. On reflection, I prefer to keep calling it the step number. 

In case its of any interest, let me just say that from my understanding an "iterate" is something countable (as the set of integers [0... n_timesteps] is). That was my original motivation for the suggestion. The step number is also an accumulation of the incrementation of the step count, which means that you iterate (v) to get the iterate (n). English is weird.